Swiften/Elements/MIXParticipant.h

```diff
@@ -37,7 +37,7 @@ namespace Swift {
                 return jid_;
             }
 
-            void setJID(const JID& jid) {
+            void setJID(const boost::optional<JID>& jid) {
```

Comment:
This can probably be just JID too. I mean the reasoning for JID and Nickname being optional or not is probably the same.